src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -231,6 +231,14 @@ GcsAutoscalerStateManager::GetAggregatedResourceLoad() const {
   return aggregate_load;
 };
 
+void GcsAutoscalerStateManager::Initialize(const GcsInitData &gcs_init_data) {
```

Comment:
Looks like GcsPlacementGroupManager also initializes from the gcs_init_data, so I guess there could be inconsistency before the first report from syncer to populate the actual node resources: 

- GcsAutoscaeler would proabbly think all nodes are empty after this `Initalize`
- But GcsPlacementGroupManager thinks some placement groups are already scheduled on the worker nodes (which they are) 

Unless we persist the cached autoscaler as part of init_data, but I feel this is probably fine.

Reply:
GcsAutoscalerStateManager does not maintain placement group state, though? It uses `GcsPlacementGroupManager` as the source of truth

Reply:
So any reported state will combine the two

Reply:
yes, but GcsAutoscalerStateManager has the available resources info. But I think with PGmanager trying to rescheudle PGs, it should be synced before GCS reports to Autoscaler.